<?xml version="1.0"?>

<robot name="mrobot" xmlns:xacro="http://www.ros.org/wiki/xacro">

	<xacro:include filename="$(find avp_mapping)/simulation/robot/robot_body.urdf.xacro" />
    <!--调用xarco定义的宏-->
    <robot_body/>


	<link name="camera0_link">
        <visual>
          <origin xyz=" 0 0 0 " rpy="0 0 0" />
          <geometry>
          <sphere radius="0.01" />
          </geometry>
          <material name="black">
          <color rgba="0 0 0 0.95"/>
          </material>
        </visual>
    </link>

	<joint name="realsense_frame_joint" type="fixed">
		<origin xyz="0 0 2.0" rpy="0 0.25 0" />
		<parent link="base_link"/>
		<child link="camera0_link"/>
	</joint>

  <link name="camera1_link">
      <visual>
        <origin xyz=" 0 0 0 " rpy="0 0 0" />
        <geometry>
        <sphere radius="0.01" />
        </geometry>
        <material name="black">
        <color rgba="0 0 0 0.95"/>
        </material>
      </visual>
      </link>

      <joint name="camera1_joint" type="fixed">
        <origin xyz="0.00 0 2.0" rpy="0 0.25 1.0471975511965976"/>
        <parent link="base_link"/>
        <child link="camera1_link"/>
  </joint> 


  <link name="camera2_link">
        <visual>
          <origin xyz=" 0 0 0 " rpy="0 0 0" />
          <geometry>
          <sphere radius="0.01" />
          </geometry>
          <material name="black">
          <color rgba="0 0 0 0.95"/>
          </material>
        </visual>
        </link>
  
        <joint name="camera2_joint" type="fixed">
          <origin xyz="0.00 0 2.0" rpy="0 0.25 2.0943951023931953"/>
          <parent link="base_link"/>
          <child link="camera2_link"/>
  </joint>

  <link name="camera3_link">
          <visual>
            <origin xyz=" 0 0 0 " rpy="0 0 0" />
            <geometry>
            <sphere radius="0.01" />
            </geometry>
            <material name="black">
            <color rgba="0 0 0 0.95"/>
            </material>
          </visual>
          </link>
    
          <joint name="camera3_joint" type="fixed">
            <origin xyz="0.00 0 2.0" rpy="0 0.25 3.141592653589793"/>
            <parent link="base_link"/>
            <child link="camera3_link"/>
          </joint> 

          <link name="camera4_link">
            <visual>
              <origin xyz=" 0 0 0 " rpy="0 0 0" />
              <geometry>
              <sphere radius="0.01" />
              </geometry>
              <material name="black">
              <color rgba="0 0 0 0.95"/>
              </material>
            </visual>
            </link>
      
            <joint name="camera4_joint" type="fixed">
              <origin xyz="0.00 0 2.0" rpy="0 0.25 -2.0943951023931953"/>
              <parent link="base_link"/>
              <child link="camera4_link"/>
  </joint> 


  <link name="camera5_link">
              <visual>
                <origin xyz=" 0 0 0 " rpy="0 0 0" />
                <geometry>
                <sphere radius="0.01" />
                </geometry>
                <material name="black">
                <color rgba="0 0 0 0.95"/>
                </material>
              </visual>
              </link>
        
              <joint name="camera5_joint" type="fixed">
                <origin xyz="0.00 0 2.0" rpy="0 0.25 -1.0471975511965976 "/>
                <parent link="base_link"/>
                <child link="camera5_link"/>
  </joint> 



  <gazebo reference="camera0_link">
    <sensor type="camera" name="camera0">
      <update_rate>30.0</update_rate>
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>800</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <!-- Noise is sampled independently per pixel on each frame.
               That pixel's noise value is added to each of its color
               channels, which at that point lie in the range [0,1]. -->
          <mean>0.0</mean>
          <stddev>0.0001</stddev>
        </noise>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>camera0</cameraName>
        <imageTopicName>color/image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera0_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>

  <gazebo reference="camera1_link">
    <sensor type="camera" name="camera1">
      <update_rate>30.0</update_rate>
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>800</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <!-- Noise is sampled independently per pixel on each frame.
               That pixel's noise value is added to each of its color
               channels, which at that point lie in the range [0,1]. -->
          <mean>0.0</mean>
          <stddev>0.001</stddev>
        </noise>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>camera1</cameraName>
        <imageTopicName>color/image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera1_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>

  <gazebo reference="camera2_link">
    <sensor type="camera" name="camera2">
      <update_rate>30.0</update_rate>
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>800</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <!-- Noise is sampled independently per pixel on each frame.
               That pixel's noise value is added to each of its color
               channels, which at that point lie in the range [0,1]. -->
          <mean>0.0</mean>
          <stddev>0.001</stddev>
        </noise>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>camera2</cameraName>
        <imageTopicName>color/image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera2_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>

  <gazebo reference="camera3_link">
    <sensor type="camera" name="camera3">
      <update_rate>30.0</update_rate>
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>800</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <!-- Noise is sampled independently per pixel on each frame.
               That pixel's noise value is added to each of its color
               channels, which at that point lie in the range [0,1]. -->
          <mean>0.0</mean>
          <stddev>0.001</stddev>
        </noise>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>camera3</cameraName>
        <imageTopicName>color/image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera3_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>


  <gazebo reference="camera4_link">
    <sensor type="camera" name="camera4">
      <update_rate>30.0</update_rate>
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>800</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <!-- Noise is sampled independently per pixel on each frame.
               That pixel's noise value is added to each of its color
               channels, which at that point lie in the range [0,1]. -->
          <mean>0.0</mean>
          <stddev>0.001</stddev>
        </noise>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>camera4</cameraName>
        <imageTopicName>color/image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera4_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>

  <gazebo reference="camera5_link">
    <sensor type="camera" name="camera5">
      <update_rate>30.0</update_rate>
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>800</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <!-- Noise is sampled independently per pixel on each frame.
               That pixel's noise value is added to each of its color
               channels, which at that point lie in the range [0,1]. -->
          <mean>0.0</mean>
          <stddev>0.001</stddev>
        </noise>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>camera5</cameraName>
        <imageTopicName>color/image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera5_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>




</robot>